pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -515,7 +515,7 @@ func addEndpointSubset(subsets []v1.EndpointSubset, pod *v1.Pod, epa v1.Endpoint
 			Ports:     []v1.EndpointPort{epp},
 		})
 		readyEps++
-	} else {
+	} else if shouldPodBeInEndpoints(pod) {
```

Comment:
Shouldn't be in this PR. My tig macro picks up the commit at the top of FETCH_HEAD sometimes and rewrites it....